pkg/arm/data/azuredeploy.json

```diff
@@ -29,20 +28,19 @@
             "properties": {
                 "addressSpace": {
                     "addressPrefixes": [
-                        "10.0.0.0/8"
+                        "{{ .ContainerService.Properties.NetworkProfile.VnetCIDR | js }}"
                     ]
                 },
                 "subnets": [
                     {
                         "name": "default",
                         "properties": {
-                            "addressPrefix": "10.0.0.0/24"
+                            "addressPrefix": "{{ (index .ContainerService.Properties.AgentPoolProfiles 0).SubnetCIDR | js }}"
```

Comment:
It's fine to use the first agentPoolProfile subnet CIDR for now since we enforce that all of them need to be the same but in https://github.com/openshift/openshift-azure/issues/537 we should probably link this to save some time for the assignee.